lumigator/backend/backend/services/jobs.py

```diff
@@ -57,25 +60,167 @@
 JobSpecificRestrictedConfig = type[JobEvalConfig | JobInferenceConfig]
 
 
+# The end result should be that InferenceJobConfig is actually JobInferenceConfig
+# (resp. Eval)
+# For the moment, something will convert one into the other, and we'll decide where
+# to put this. The jobs should ideally have no dependency towards the backend.
+
+
+class JobDefinition(ABC):
+    """Abstract base class for jobs.
+
+    Attributes:
+    ----------
+    command : str
+        The command to execute the job.
+    pip_reqs : str, optional
+        Path to a requirements file specifying dependencies (default: None).
+    work_dir : str, optional
+        Working directory for the job (default: None).
+    ray_worker_gpus_fraction : float
+        Fraction of a GPU allocated per Ray worker.
+    ray_worker_gpus : float
+        Number of GPUs allocated per Ray worker.
+    config_model : BaseModel
+        Pydantic model representing job-specific configuration.
+
+    """
+
+    command: str
+    pip_reqs: str | None
+    work_dir: str | None
+    ray_worker_gpus_fraction: float
+    ray_worker_gpus: float
+    config_model: BaseModel
+
+    # This should end up not being necessary, since we'd expose the whole job config
+    @abstractmethod
+    def generate_config(self, request: JobCreate, record_id: UUID, dataset_path: str, storage_path: str) -> BaseModel:
+        """Generates
+
+        Parameters
+        ----------
+        request : JobCreate
+            Non-job-specific parameters for job creation
+        record_id : UUID
+            Job ID assigned to the job
+        dataset_path : str
+            S3 path for the input dataset
+        storage_path : str
+            S3 where the backend will store the results from the job
+
+        Returns:
+        -------
+        generate_config : BaseModel
+            A job-specific pydantic model that will be sent to the Ray job instance
+        """
+        pass
+
+    @abstractmethod
+    def store_as_dataset(self) -> bool:
+        """Returns:
+        -------
+        store_as_dataset : bool
+            Whether the results should be stored in an S3 path
+        """
+        pass
+
+    def __init__(self, command, pip_reqs, work_dir, ray_worker_gpus_fraction, ray_worker_gpus, config_model):
+        self.command = command
+        self.pip_reqs = pip_reqs
+        self.work_dir = work_dir
+        self.ray_worker_gpus_fraction = ray_worker_gpus_fraction
+        self.ray_worker_gpus = ray_worker_gpus
+        self.config_model = config_model
+
+
+class JobDefinitionInference(JobDefinition):
```

Comment:
Does it make sense for these new classes to be in a different file? this jobs.py file is getting busy

Reply:
Wait for next PR..... ^_^